pkg/predicate/predicate.go

```diff
@@ -268,3 +270,16 @@ func (o or) Generic(e event.GenericEvent) bool {
 	}
 	return false
 }
+
+// LabelSelectorPredicate takes in a LabelSelector as a parameter and returns predicate functions. The labels of
+// events that are contained as parameters in these predicate functions are compared against the instantiated
+// Selector object and reconciliation of the event will only take place if the labels match.
+func LabelSelectorPredicate(s metav1.LabelSelector) (Funcs, error) {
```

Comment:
Please change this to return `(Predicate, error)`. The `Funcs` are an implementation detail ppl should not rely on.